sw/host/opentitanlib/src/test_utils/lc_transition.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::rc::Rc;
+use std::time::Duration;
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use crate::app::TransportWrapper;
+use crate::chip::boolean::MultiBitBool8;
+use crate::dif::lc_ctrl::{
+    DifLcCtrlState, LcBit, LcCtrlReg, LcCtrlStatusBit, LcCtrlTransitionCmdBit,
+    LcCtrlTransitionCtrlBit,
+};
+use crate::impl_serializable_error;
+use crate::io::jtag::{Jtag, JtagTap};
+
+/// Errors related to performing an LcTransition.
+#[derive(Error, Debug, Deserialize, Serialize)]
+pub enum LcTransitionError {
+    #[error("LC controller not ready to perform an LC transition (status: {0:x}).")]
+    LcCtrlNotReady(u32),
+    #[error("Maximum LC transitions reached (transition count: {0}).")]
+    MaxTransitionCountExceeded(u32),
+    #[error("LC transition mutex was already claimed.")]
+    MutexAlreadyClaimed,
+    #[error("Failed to claim LC transition mutex.")]
+    FailedToClaimMutex,
+    #[error("LC transition target programming failed (target state: {0:x}).")]
+    TargetProgrammingFailed(u32),
+    #[error("LC transition failed.")]
+    TransitionFailed,
+    #[error("Bad post transition LC state: {0:x}.")]
+    BadPostTransitionState(u32),
+    #[error("Generic error {0}")]
+    Generic(String),
+}
+impl_serializable_error!(LcTransitionError);
+
+pub fn trigger_lc_transition(
+    transport: &TransportWrapper,
+    jtag: Rc<dyn Jtag>,
+    target_lc_state: DifLcCtrlState,
+    use_external_clk: bool,
+    reset_delay: Duration,
+) -> Result<()> {
+    const MAX_LC_TRANSITIONS: u32 = 24;
```

Comment:
Maybe this is fixed for Earlgrey, but it would be more robust to check the `STATUS.TRANSITION_COUNT_ERROR` bit for transition count errors